src/include/stir/FilePath.h

```diff
@@ -0,0 +1,187 @@
+#define __stir_FILEPATH_H__
+
+/*!
+  \file
+  \ingroup buildblock
+
+  \brief Declaration of class stir::FilePath
+  This is a class implementing basic filesytem functionality. Parts of this class were
+  copied from Functions for filename manipulations.
+
+  Maximum effort was put in order to be cross platform.
+
+  \author Nikos Efthimiou
+  \author Kris Thielemans
+  \author PARAPET project
+
+*/
+
+#include "stir/common.h"
+#include <stdio.h>
+#include <iostream>
+#include <string>
+#include <boost/format.hpp>
+
+
+#if defined(__OS_WIN__)
+	#include <windows.h>
+	 #include<sys/types.h> // required for stat.h
+	#include<direct.h>
+#else
+	#include <unistd.h>
+#endif
+
+#include <sys/stat.h>
+START_NAMESPACE_STIR
+
+/*!
+ * \ingroup buildblock
+ *
+ * \brief The FilePath class
+ *
+ * \par Description
+ * This class provides basic filesystem support for STIR. It is build in a similar manner to the functions in the utilities but extends with support to create folders, check if folder exists and is writable.
+ *
+ * The old pathname operations are supported.
+ */
+
+class FilePath
+{
+public:
+    FilePath();
+
+    //!
+    //! \brief FilePath
+    //! \param __str Name of path
+    //! \param _check Perform checks to the __str
+    //!
+    //! If the initial path exists in the disk _check should be left true.
+    //! It will check that the path is writtable
+    //! If you want to initialize to an arbitary string then set _check false.
+    FilePath(const std::string &__str, bool _check = true);
+
+    ~FilePath()
+    {}
+
+    //! Returns true if my_string points to a directory
+    bool is_directory() const;
+    //! Returns true if my_string points to a regular file
+    //! Returns true if the path is writable. 
+	//! \warning As far as I understand is only a *NIX feature. Write permissions do not exist in this level at Windows. 
+    bool is_regular_file() const;
+    bool is_writable() const;
+    //! Returns true if the path is absolute
+    //! This funtion has been copied from the Functions for filename manipulations
+    static bool is_absolute(const std::string& _filename_with_directory);
+    //! Returns true if the path already exists
+    static bool exist(std::string s);
+    //! Returns the current / working directory
+    static std::string get_current_working_directory();
+    //! Create a new folder,by FilePath, and return its path
+    FilePath append(const FilePath& p);
+    //! \brief Create a new folder,by string, and return its path
+    //! \details This functions creates appends path p to my_string.
+    //! It supports multiple new folder levels.It will try to avoid
+    //! errors as permissions and non existing root path.
+    FilePath append(const std::string& p);
+    //! Append the separator
+    static void append_separator(std::string& s);
+    //! If path is no absolute then prepend the p string
+    //! This funtion has been copied from the Functions for filename manipulations
```

Comment:
remove the `!`. no point in having that appear in the doxygen-generated files. Misspelling `function`